File: Data Structures & Algorithms/duplicate-integer/submission-0.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        map<int, int> mp;
        int n = nums.size();
        for(int i = 0; i < n ; i++){
            if(mp.find(nums[i]) != mp.end()){
                return 1;
            }else{
                mp[nums[i]] += 1;
            }
        }
        return 0;

    }
};

File: Data Structures & Algorithms/find-target-in-rotated-sorted-array/submission-0.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        l, r = 0, len(nums) - 1
        while l < r:
            m = (l + r) // 2
            if nums[m] < nums[r]:
                r = m
            else:
                l = m + 1
        
        pivot = l 
        l, r = 0, len(nums) - 1
        if target <= nums[r] and target >= nums[pivot]:
            l = pivot
        else:
            r = pivot - 1

        res = -1
        while l <= r:
            m = (l + r) // 2
            
            if target == nums[m]:
                res = m
                break

            if nums[m] > target:
                r = m - 1
            else:
                l = m + 1

        return res


